beacon-chain/sync/validate_beacon_blocks_test.go

```diff
@@ -63,14 +62,21 @@ func TestValidateBeaconBlockPubSub_InvalidSignature(t *testing.T) {
 	msg.Block.ParentRoot = bRoot[:]
 	msg.Block.Slot = 1
 	msg.Block.ProposerIndex = proposerIdx
-	msg.Signature = bytesutil.PadTo([]byte("fake"), fieldparams.BLSSignatureLength)
+	badPrivKeyIdx := proposerIdx - 1 // We generate a valid signature from a wrong private key which fails to verify
+	if badPrivKeyIdx < 0 {
+		badPrivKeyIdx += 2
+	}
```

Comment:
`proposerIdx` is deterministic, you could just say `badPrivKeyIdx = proposerIdx +1`  ?

Reply:
I wasn't sure if it was deterministic so I added a check just in case. So it can definitely be simplified,